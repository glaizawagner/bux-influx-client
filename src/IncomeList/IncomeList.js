import React, { Component } from 'react';
import IncomeItem from '../IncomeItem/IncomeItem';
import BuxInfluxContext from  '../BuxInfluxContext'

export default class IncomeList extends Component {
    static defaultProps = {
        income: []
      }

    static contextType = BuxInfluxContext;

    render() {
        const { income } = this.context;
        return(
            <section className='IncomeList'>
                <h2>Income</h2>
                <ul className="IncomeList__list" aria-live='polite'>
                    {income.map((income,i) => 
                        <IncomeItem 
                            key={i} 
                            iid={i}
                            {...income} 
                        />
                    )}
                </ul>
            </section>    
        );
    }
}

