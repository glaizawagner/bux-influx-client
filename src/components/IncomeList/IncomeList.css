.incName {
    color: #4c924c;
    text-transform: uppercase;
    display: block;
    text-align: center;
    font-size: 2.5rem;
    font-weight: 900;
    margin-top: 3rem;
    word-spacing: 2em;
    border-top: 2px solid black;
    border-bottom: 2px solid black;
    padding: 0;
    letter-spacing: 1rem;
}

.IncomeList__list {
    text-align: left;
    margin: 2% 5%;
    list-style: none;
    /* border-top: 2px solid black;
    border-bottom: 2px solid black; */
}

@media only screen and (min-width: 37.5em) and (max-width: 56.25em) {
    .incName {
        margin-top: 3rem;
        margin-left: 10%;
        margin-right: 10%;
    }

    .IncomeList__list{
        margin: 1% 10%;
    }
}

@media only screen and (min-width: 56.25em) and (max-width: 75em) {
    .incName {
        margin-top: 3rem;
        margin-left: 10%;
        margin-right: 10%;
    }

    .IncomeList__list {
        margin: 1% 10%;
    }
}

@media only screen and (min-width: 75em) and (max-width: 105em) {
    .incName {
        margin-top: 3rem;
        margin-left: 20%;
        margin-right: 20%;
    }

    .IncomeList__list {
        margin: 1% 20%;
    }
}