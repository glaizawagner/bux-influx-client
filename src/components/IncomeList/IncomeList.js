/* eslint-disable no-unused-expressions */
import React, { Component } from 'react';
import IncomeItem from '../../components/IncomeItem/IncomeItem';
import BuxinfluxContext from  '../../contexts/BuxinfluxContext'
import helpers from '../../components/helpers/helpers';
import './IncomeList.css';

export default class IncomeList extends Component {
    static defaultProps = {
        income: [],
    }

    static contextType = BuxinfluxContext;

    componentDidMount() {
        // this.context.clearError();
    }

    renderIncome(){
        const { income, created } = this.context;

        return (
                <section className='IncomeList'>
               <span className="incName">Income</span>
               <ul className="IncomeList__list" aria-live='polite'>
                    {income.map((income,i) => 
                        (helpers.formatDate(income.date_created) === helpers.formatDate(created)) 
                        ? <IncomeItem 
                            key={i} 
                            iid={i} 
                            user_id ={this.context.currentUser} 
                            {...income} 
                            // toggleEditing={() => this.toggleIncomeEditing(i)}
                            // onChange={this.handleIncomeUpdate}
                            /> 
                        : ''
                    )}
                   
           </ul>
        </section>   
        );
    }

    render() {
        const { error } = this.context
        return(
            <div className='IncomeList'>
               { error 
                ? <p className='red'> There was an error, try again</p>
                : this.renderIncome()
                }
            </div>    
        );
    }
}