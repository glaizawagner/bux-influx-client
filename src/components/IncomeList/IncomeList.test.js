import React from 'react';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer'
import IncomeList from './IncomeList';
import BuxinfluxContext from '../../contexts/BuxinfluxContext';

const contextValue = {
    income: [
        {
            iid: 1,
            date_created: '2019-11-12T16:28:32.615Z',
            type: 'inc',
            description: 'Salary',
            value: '3500.00',
            user_id: 1
        },
        {
            iid: 2,
            date_created: '2019-11-12T16:28:32.615Z',
            type: 'inc',
            description: 'Other Income',
            value: '3500.00',
            user_id: 2
        },
    ]
}

describe('Income List  component', () => {
    it('renders UI expected', () => {
        const tree = renderer
        .create(<BuxinfluxContext.Provider value={contextValue}>
                <IncomeList />
                </BuxinfluxContext.Provider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
    
    });
    
    // enzyme doesn't yet support React.createContext
    it.skip('renders Income in ul for each in array', () => {
        const props = {
            match: {
                params: {
                    eid: 'THIS_INCOME_ID'
                }
            }
        }

        const ul = (<IncomeList {...props} />, contextValue)
            .find('ul')
            expect(toJson(ul)).toMatchSnapshot()
    })

});