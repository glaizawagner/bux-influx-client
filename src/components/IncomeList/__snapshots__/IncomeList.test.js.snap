// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Income List  component renders UI expected 1`] = `
<div
  className="IncomeList"
>
  <section
    className="IncomeList"
  >
    <span
      className="incName"
    >
      Income
    </span>
    <ul
      aria-live="polite"
      className="IncomeList__list"
    >
      <li
        className="IncomeItem"
      >
        <div
          className="IncomeItem__row"
        >
          <span
            className="items"
          >
            Nov 12, 2019
          </span>
          <span
            className="items"
          >
            Salary
          </span>
          <span
            className="items"
          >
            3500.00
          </span>
          <button
            className="IncomeItem__description"
            onClick={[Function]}
          >
            <svg
              className="timesIcon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<path d=\\"M256 8C119 8 8 119 8 256s111 248 248 248s248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200s-89.5 200-200 200zm101.8-262.2L295.6 256l62.2 62.2c4.7 4.7 4.7 12.3 0 17l-22.6 22.6c-4.7 4.7-12.3 4.7-17 0L256 295.6l-62.2 62.2c-4.7 4.7-12.3 4.7-17 0l-22.6-22.6c-4.7-4.7-4.7-12.3 0-17l62.2-62.2l-62.2-62.2c-4.7-4.7-4.7-12.3 0-17l22.6-22.6c4.7-4.7 12.3-4.7 17 0l62.2 62.2l62.2-62.2c4.7-4.7 12.3-4.7 17 0l22.6 22.6c4.7 4.7 4.7 12.3 0 17z\\" fill=\\"currentColor\\"/>",
                }
              }
              focusable={false}
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "transform": "rotate(360deg)",
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            />
          </button>
        </div>
      </li>
      <li
        className="IncomeItem"
      >
        <div
          className="IncomeItem__row"
        >
          <span
            className="items"
          >
            Nov 12, 2019
          </span>
          <span
            className="items"
          >
            Other Income
          </span>
          <span
            className="items"
          >
            3500.00
          </span>
          <button
            className="IncomeItem__description"
            onClick={[Function]}
          >
            <svg
              className="timesIcon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<path d=\\"M256 8C119 8 8 119 8 256s111 248 248 248s248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200s-89.5 200-200 200zm101.8-262.2L295.6 256l62.2 62.2c4.7 4.7 4.7 12.3 0 17l-22.6 22.6c-4.7 4.7-12.3 4.7-17 0L256 295.6l-62.2 62.2c-4.7 4.7-12.3 4.7-17 0l-22.6-22.6c-4.7-4.7-4.7-12.3 0-17l62.2-62.2l-62.2-62.2c-4.7-4.7-4.7-12.3 0-17l22.6-22.6c4.7-4.7 12.3-4.7 17 0l62.2 62.2l62.2-62.2c4.7-4.7 12.3-4.7 17 0l22.6 22.6c4.7 4.7 4.7 12.3 0 17z\\" fill=\\"currentColor\\"/>",
                }
              }
              focusable={false}
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "transform": "rotate(360deg)",
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            />
          </button>
        </div>
      </li>
    </ul>
  </section>
</div>
`;
